scintillometry/base/base.py

```diff
@@ -0,0 +1,160 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import operator
+import numpy as np
+import astropy.units as u
+
+
+class TaskBase(object):
```

Comment:
Agreed that values should be passed if `TaskBase` methods need them (otherwise it just gets confusing).  I've made `shape`, `sample_rate` and `dtype` positional arguments to `__init__` - this way we're guaranteed for eg. the `sample_shape` property to work.